app/models/piece.rb

```diff
@@ -124,13 +124,21 @@ def never_moved?
     updated_at == created_at
   end
 
+  def active_player
+    if self.game.move_counter % 2 == 0
+      return self.game.white_user
+    end
+    return self.game.black_user
+  end
+
   def move_to!(x,y)
-    return false unless valid_move?(x,y)
-    capture_piece_at(x,y)
+    return false unless valid_move?(x, y) && self.active_player.id == self.user_id
```

Comment:
If it works locally, then it must be checked somewhere else and you can ignore my comment.  This line by itself would let either white or black move white's piece when it's white's turn, and vice versa.  But there must be a check elsewhere, so you should be good.  👍